data/scenarios/Challenges/blender.yaml

```diff
@@ -0,0 +1,226 @@
+version: 1
+name: Fortress infiltration
+author: Karl Ostmo
+description: |
+  Navigate enemy-infested passages of Lambda Fortress to retrieve the prize.
+creative: false
+objectives:
+  - teaser: Get amulet
+    goal:
+      - |
+        `grab` the Amulet of Yoneda from the northwest sanctum while
+        timing your passage carefully to avoid Side Effects (X) on patrol.
+      - |
+        To unlock a red door, `drill` it with the "door key" equipped.
+    condition: |
+      as base {has "Amulet of Yoneda"}
+    prerequisite:
+      not: get_caught
+  - id: get_caught
+    teaser: Apprehended
+    goal:
+      - Got caught by the patrolling bots
+    hidden: true
+    optional: true
+    condition: |
+      // run "scenarios/Challenges/_blender/apprehension-checker.sw"
+      def hasMetBase = \r.
+          let basename = "base" in
+          x <- as r {whoami};
+          if (x == basename) {
+              return false;
+          } {
+              mr0 <- as r {meet};
+              case mr0
+                (\_. return false)
+                (\bot. name <- as bot {whoami}; return $ name == basename);
+          };
+          end;
+
+      /**
+      Iterates sequentially until
+      encountering an invalid robot index.
+
+      Distinguishes system bots from the base by name.
+      Returns true if a bot has "met" the base.
+      */
+      def anyHasMetBase : int -> cmd bool = \idx.
+          
+          try {
+              bot <- robotnumbered idx;
+              intermediate <- hasMetBase bot;
+              let foo = intermediate in
+              let newIdx = idx + 1 in
+              recursiveResult <- anyHasMetBase newIdx;
+              return $ foo || recursiveResult;
+          } {
+            // Terminates the recursion on the
+            // lowest index at which a robot does not exist
+            return false;
+          };
+          end;
+
+      anyHasMetBase 1;
+robots:
+  - name: base
+    dir: [1, 0]
+    devices:
```

Comment:
Can we get a `compass` please?  I solved it using a key handler + making a driving mode so I could do it manually, but I had to use the version you evidently prefer, of having up mean `move` and left/right mean `turn`.  However, my brain finds that profoundly confusing and I had to restart a few times since I messed up the turning.  I prefer having e.g. up mean `turn north; move`, left mean `turn west; move` and so on.